### PR TITLE
Add libxc 5.0.0

### PR DESCRIPTION
**Pull request by haampie**
With experimental CUDA support and some patches to make it compile.

Currently +shared and +cuda conflict, this has to be fixed upstream.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libxc/package.py [hunk 1]
```diff
@@ -6,19 +6,28 @@
...
     url      = "https://www.tddft.org/programs/libxc/down.php?file=2.2.2/libxc-2.2.2.tar.gz"
 
+    version('5.0.0', sha256='1cdc57930f7b57da4eb9b2c55a50ba1c2c385936ddaf5582fee830994461a892')
     version('4.3.2', sha256='bc159aea2537521998c7fb1199789e1be71e04c4b7758d58282622e347603a6f')
     version('4.2.3', sha256='02e49e9ba7d21d18df17e9e57eae861e6ce05e65e966e1e832475aa09e344256')
     version('3.0.0', sha256='5542b99042c09b2925f2e3700d769cda4fb411b476d446c833ea28c6bfa8792a')
     version('2.2.2', sha256='6ca1d0bb5fdc341d59960707bc67f23ad54de8a6018e19e02eee2b16ea7cc642')
     version('2.2.1', sha256='ade61c1fa4ed238edd56408fd8ee6c2e305a3d5753e160017e2a71817c98fd00')
 
+    variant('shared', default=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libxc/package.py]: Needs a description

File: var/spack/repos/builtin/packages/libxc/package.py [hunk 2]
```diff
@@ -6,19 +6,28 @@
...
+
+    conflicts('+shared +cuda', msg='Only ~shared supported with +cuda')
+    conflicts('+cuda', when='@:4.3', msg='CUDA support only in libxc 5.0.0 and above')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libxc/package.py]:
```suggestion
    conflicts('+cuda', when='@:4', msg='CUDA support only in libxc 5.0.0 and above')
```

File: var/spack/repos/builtin/packages/libxc/package.py [hunk 3]
```diff
@@ -81,8 +91,25 @@ def setup_build_environment(self, env):
...
+
     def configure_args(self):
-        args = ['--enable-shared']
+        args = []
+
+        if '+shared' in self.spec:
+            args.append('--enable-shared')
+
+        if '+cuda' in self.spec:
+            args.append('--enable-cuda')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libxc/package.py]: Can these be explicitly disabled in else clauses?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by adamjstewart** [comment]
You'll need to resolve the merge conflicts now that #17781 has been merged. Let me know if you need help.

**Comment by haampie** [comment]
Thanks for the review!

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**